src/main/java/com/es/phoneshop/model/parser/QuantityParser.java

```diff
@@ -0,0 +1,29 @@
+package com.es.phoneshop.model.parser;
+
+import com.es.phoneshop.exception.CustomParseException;
+import lombok.Setter;
+
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.Locale;
+
+@Setter
+public class QuantityParser implements Parser<Integer> {
+
+    private Locale locale = Locale.ENGLISH;
+
+    @Override
+    public Integer parse(String str) throws CustomParseException {
+        NumberFormat numberFormat = NumberFormat.getInstance(locale);
+        int quantity;
+        try {
+            quantity = numberFormat.parse(str).intValue();
+        } catch (ParseException exception) {
+            throw new CustomParseException("Quantity was not a number");
```

Comment:
i agree, that rethrowing custom exception was a bad idea (i've fixed this) , but i think if we use map for errors in this validator class, it will not be comfortable to use map when we parse only one quantity (for example when we add product to cart in pdp) or if we add a page that does not update anything if at least one quantity value is invalid; plus , i think it will be a problem to create accordance between errors and products (quantities of which were invalid) : i think validator should not know about products id, so we will probably have to create one more map with products id and error messages

Reply:
the idea is that every validation that you do can update your errorMap
no matter where you're validating, the flow is the same for all the pages
and I didn't notice any QuantityParser inside the PDP, but for sure it should be there as well

Reply:
storing id-s in this map is also fine for PDP